pkg/services/vmoperator/vmopmachine.go

```diff
@@ -270,6 +283,76 @@ func (v *VmopMachineService) ReconcileNormal(ctx context.Context, machineCtx cap
 	return false, nil
 }
 
+const (
+	maxNameLength = 63
+)
+
+// Note: Inlining these functions from sprig to avoid introducing a dependency.
+var nameTemplateFuncs = map[string]any{
+	"trimSuffix": func(a, b string) string { return strings.TrimSuffix(b, a) },
+	"trunc": func(c int, s string) string {
+		if c < 0 && len(s)+c > 0 {
+			return s[len(s)+c:]
+		}
+		if c >= 0 && len(s) > c {
+			return s[:c]
+		}
+		return s
+	},
+}
+
+var nameTpl = template.New("name generator").Funcs(nameTemplateFuncs).Option("missingkey=error")
+
+// virtualMachineObjectKey returns the object key of the VirtualMachine.
+// Part of this is generating the name of the VirtualMachine based on the naming strategy.
+func virtualMachineObjectKey(machineName, machineNamespace string, namingStrategy *vmwarev1.VirtualMachineNamingStrategy) (*client.ObjectKey, error) {
+	name, err := GenerateVirtualMachineName(machineName, namingStrategy)
+	if err != nil {
+		return nil, err
+	}
+
+	return &client.ObjectKey{
+		Namespace: machineNamespace,
+		Name:      name,
+	}, nil
+}
+
+// GenerateVirtualMachineName generates the name of a VirtualMachine based on the naming strategy.
+func GenerateVirtualMachineName(machineName string, namingStrategy *vmwarev1.VirtualMachineNamingStrategy) (string, error) {
```

Comment:
should this func return a book in case the name is truncated, so we can eventually give a warning in the web hook

Reply:
I would only implement that once/if we decide to return a warning.

Pleaes note that if we return a warning in the VSphereMachineTemplate webhook the logs of our Cluster topology controller will be full with these warnings. I'm not sure if we really want that (especially as that's an entirely expected scenario and nothing concerning for folks reading the logs)

Reply:
Also not sure how to do it. We only have the VSphereMachineTemplate webhook, at this time we don't know if VirtualMachine names will be later truncated in the controller (we just run through generation here with a test Machine name)

Reply:
ACK, if it is not already there, then let's add a note on the template field that clarifies that generated names must be valid K8s node names and that CAPV automatically truncates names exceeding max Lenght